### PR TITLE
proto serialize logical plans

### PR DESCRIPTION
**Pull request by universalmind303**
this uses the datafusion `Extension` as a means to transmit the custom nodes. 

In the process, we convert our logicalPlan to datafusion extension nodes.

for example
`LogicalPlan(DDL(CreateTable))` -> `DfLogicalPlan(Extension(CreateTable))`.

### CODE REVIEW

File: crates/sqlexec/src/session.rs [hunk 1]
```diff
@@ -231,10 +232,20 @@ impl Session {
...
+        if let DfLogicalPlan::Extension(extension) = &plan {
+            match extension.node.name() {
+                "CreateTable" => {
+                    let create_table = extension.node.as_any().downcast_ref::<CreateTable>();
+                    return self.create_table(create_table.unwrap().clone()).await;
+                }
+                _ => {}
+            }
+        }
+
```
Comment (universalmind303) [inline crates/sqlexec/src/session.rs:248]: we want to handle the extensions using our session context, not datafusions, so we need to handle those cases here.

File: crates/sqlexec/Cargo.toml [hunk 2]
```diff
@@ -32,6 +32,10 @@ tokio-postgres = "0.7.8"
 once_cell = "1.18.0"
 url.workspace = true
 parking_lot = "0.12.1"
+serde = { workspace = true }
+
+[dependencies.sqlparser]
+features = ["serde"]
```
Comment (scsmithr) [inline crates/sqlexec/Cargo.toml]: What's this for?

**Review by scsmithr — COMMENTED** [review]
lgtm, definitely feels like something we can work off of.

**Review by scsmithr — APPROVED** [review]

File: crates/sqlexec/Cargo.toml (hunk 2 above)
Comment (universalmind303) [reply crates/sqlexec/Cargo.toml]: was looking into something & forgot to remove this.

**Review by vrongmeal — APPROVED** [review]

**Pull request merged**